### PR TITLE
Display fines using FOLIO model

### LINKED ISSUE #830: Hook up fees and fines display
Building on https://github.com/sul-dlss/mylibrary/pull/828:

- [x] Translations of FOLIO feeFineTypes into patron-visible labels
~~`sequence` (used to... figure out if the patron recently paid?)~~ broken out into https://github.com/sul-dlss/mylibrary/issues/832
- [x] is `owed` the right field? fixture fines didn't seem to have it
- [x] is `dateCreated` the right field to replace `billDate`? fixture fines didn't seem to have that populated
- [x] should there be an `itemId` attached to fees + fines? the API docs say it's required, but the fixtures didn't seem to have it

### PR DESCRIPTION
**Pull request by marlo-longley**
Closes #830

### CODE REVIEW

File: spec/models/folio/fine_spec.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  end
+
+  describe '#patron_key' do
+    subject(:patron_key) { fine.patron_key }
+
+    context 'when the fine is not a proxy fine' do
+      it { expect(patron_key).to eq 'd7b67ab1-a3f2-45a9-87cc-d867bca8315f' }
+    end
+
+    # TODO
```
Comment (jcoyne) [inline spec/models/folio/fine_spec.rb:56]: Would you be able to complete this now, remove it or put a Githhub issue number in here to track this work?

File: spec/views/fines/index.html.erb_spec.rb [hunk 2]
```diff
@@ -3,48 +3,114 @@
...
+      assign(:checkouts, checkouts)
+      without_partial_double_verification do
+        allow(view).to receive(:patron_or_group).and_return(patron)
+        allow(view).to receive(:patron).and_return(patron)
+        allow(fine).to receive(:to_partial_path).and_return('fines/fine')
+        allow(checkouts).to receive(:recalled?).and_return(false)
+      end
+    end
+
+    it 'show the fined item author' do
```
Comment (jcoyne) [inline spec/views/fines/index.html.erb_spec.rb:102]:
```suggestion
    it 'shows the fined item author' do
```

File: spec/views/fines/index.html.erb_spec.rb [hunk 3]
```diff
@@ -3,48 +3,114 @@
...
 
-    it 'does not show the shared computer payment alert' do
+    before do
+      assign(:fines, fines)
+      assign(:checkouts, checkouts)
+      without_partial_double_verification do
+        allow(view).to receive(:patron_or_group).and_return(patron)
+        allow(view).to receive(:patron).and_return(patron)
+        allow(fine).to receive(:to_partial_path).and_return('fines/fine')
+        allow(checkouts).to receive(:recalled?).and_return(false)
```
Comment (jcoyne) [inline spec/views/fines/index.html.erb_spec.rb:98]: Is there a reason we aren't verifying these doubles?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: spec/models/folio/fine_spec.rb (hunk 1 above)
Comment (thatbudakguy) [reply spec/models/folio/fine_spec.rb:56]: I think this is https://github.com/sul-dlss/mylibrary/issues/873

File: spec/views/fines/index.html.erb_spec.rb (hunk 3 above)
Comment (marlo-longley) [reply spec/views/fines/index.html.erb_spec.rb:98]: Do you mean `without_partial_double_verification` ? The dumb answer is, without it the tests don't work and I get errors like
```
 1) fines/index with a FOLIO Fine shows the fined item author
     Failure/Error: allow(view).to receive(:patron_or_group).and_return(patron)
       #<ActionView::Base:0x0000000000a8c0> does not implement: patron_or_group
     # ./spec/views/fines/index.html.erb_spec.rb:95:in `block (3 levels) in <top (required)>'
     # /Users/marloelilongley/.gem/ruby/3.1.2/gems/webmock-3.18.1/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
```

If there is a different way to handle this I am open to it.

**Comment by marlo-longley** [comment]
I fixed the typo and added a link to the proxy/TODO issue. I should hopefully be doing on that this week.

File: spec/views/fines/index.html.erb_spec.rb (hunk 3 above)
Comment (jcoyne) [reply spec/views/fines/index.html.erb_spec.rb:98]: @marlo-longley yes, I understand why you need that for `view`, but I don't see why `fine` or `checkouts` would give you that.

Comment (marlo-longley) [reply spec/views/fines/index.html.erb_spec.rb]: Gotcha thanks for explaining. I just fixed this.

**Pull request merged**